src/main/java/signup_and_login/WelcomeView.java

```diff
@@ -0,0 +1,59 @@
+package signup_and_login;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class WelcomeView {
+    private JTextField usernameField;
+    private JPasswordField passwordField;
+    private final UserController controller;
+
+    public WelcomeView(UserController controller){
+        this.controller = controller;
+    }
+
+    /**
+     * Prepares and displays the graphical user interface for user registration.
+     * This method creates the necessary components such as labels, text fields, and buttons,
+     * sets up the layout, and handles the submission of user information.
+     */
+    public void prepareGUI() {
+
+        JFrame frame = new JFrame("User Registration");
+        frame.setLayout(new BorderLayout());
+
+        JPanel panel = new JPanel(new GridLayout(3, 2, 10, 10));
+        panel.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+
+        JLabel usernameLabel = new JLabel("Username:");
+        usernameField = new JTextField();
+        JLabel passwordLabel = new JLabel("Password:");
+        passwordField = new JPasswordField();
+
+        JButton submitButton = new JButton("Submit");
+        submitButton.addActionListener(e -> {
+            String username = usernameField.getText();
+            String password = new String(passwordField.getPassword());
+
+            if(!username.isEmpty() && !password.isEmpty()) {
```

Comment:
We could potentially make the error specific to the empty field (i.e. password empty! if no password or username empty! if no username). We could also have checks to make sure the username conforms to a certain standard (use regex) and the password is sufficiently complex.

Reply:
see c51ace0155cc3042f480218b441446e951fe0cf3